bcl2fastq/index_summary.py

```diff
@@ -169,21 +174,21 @@ def update_pie_chart(run_alias, known_json, unknown_json):
     pruned_count = pruned['Count'].sum()
 
     total_clusters = gsiqcetl.bcl2fastq.utility.total_clusters_for_run(
-        run_alias, index
+        run_alias, index1
```

Comment:
Why would this function still require access the pre-filtered DataFrame? The run_alias options would not include any invalid runs, so the DataFrame does not need to contain them.

Reply:
Without this change, it results in an error 
![image](https://user-images.githubusercontent.com/51453935/59789381-ce493080-929b-11e9-852b-8d0226ccc12f.png)

Reply:
That is a failed sanity check from GSI-QC-ETL. Let me investigate.

Reply:
Ahh, my past self saves me.

Currently, there are these possibilities for each lane of bcl2fastq analysis:

|Scenario| Index1 | Index2 |SampleID|Is it valid?|
| ---| --- | --- | --- | --- |
|1|ATGC|ATGC|Sample_1|Yes, dual index
|2|ATGC|NaN|Sample_1|Yes, single first index
|3|NaN|ATGC|Sample_1|Yes, dual second index
|4|NaN|NaN|NaN|Yes, the unknown indices in the lane
|5|NaN|NaN|Sample_1|No, a sample must have at least one index

Only Scenario 5 is invalid. With `index1.dropna(subset=('Index1','Index2'), thresh = 1)` we are getting rid of Scenario 4 and 5. When the total clusters are being calculated with `gsiqcetl.bcl2fastq.utility.total_clusters_for_run`, the unknown indices have been removed and the function correctly points out that those indices are needed to calculate the total clusters.

Irene's solution to this is to keep two DataFrames, which works. The issue is that now there are two very similar global DataFrames, making one more likely to use the wrong one and increasing memory footprint.

I can think of two solutions:

### Solution 1

Selectively remove runs that have at least 1 invalid library (while not removing Scenario 4 rows).  This requires a more challenging filter. It will require the use of `groupby` and `filter` functions (https://pandas.pydata.org/pandas-docs/stable/user_guide/groupby.html#).

These are the more complex and useful functions. To give you a hint, here is a way to filter for runs that have at least one library (row) in Scenario 1 (dual indexed):

`index.groupby('Run').filter(lambda x: any(~x['Index1'].isna() & ~x['Index2'].isna()))`

### Solution 2
Covert any instance of Scenario 5 into Scenario 4. In other words, if a row is invalid, it is treated as though this is an unknown index. This should be simpler to implement, but might require additional logic to deal with plotting runs that have no known indices.